socketmaster.go

```diff
@@ -30,9 +30,12 @@ func handleSignals(processGroup *ProcessGroup, c <-chan os.Signal, startTime int
 					time.Sleep(time.Duration(startTime) * time.Millisecond)
 				}
 
-				// A possible improvement woud be to only swap the
-				// process if the new child is still alive.
-				processGroup.SignalAll(syscall.SIGTERM, process)
+				if len(processGroup.set.set) > 1 {
+					processGroup.SignalAll(syscall.SIGTERM, process)
+				} else {
+					processGroup.SignalAll(syscall.SIGCHLD, nil)
```

Comment:
hmm well yeah that's make sense, maybe that work around only for my cases. Let me clean it up then.

sure i'll take a look at crank, but if i'm going to use that. this PR might not be useful for me haha